Firestore/core/test/unit/api/load_bundle_task_test.cc

```diff
@@ -0,0 +1,415 @@
+                                   });
+
+  task.SetError();
+
+  EXPECT_EQ(1, queue.pop());
+}
+
+TEST(LoadBundleTaskTest, ProgressObserveTriggers) {
+  LoadBundleTask task(CreateUserQueue());
+  BlockingQueue<int> queue;
+
+  auto handle1 = task.ObserveState(LoadBundleTaskState::InProgress,
+                                   [&](LoadBundleTaskProgress p) {
+                                     EXPECT_EQ(p, Progress(1, 5));
+                                     queue.push(1);
+                                   });
+  auto handle2 = task.ObserveState(LoadBundleTaskState::InProgress,
+                                   [&](LoadBundleTaskProgress p) {
+                                     EXPECT_EQ(p, Progress(1, 5));
+                                     queue.push(2);
+                                   });
+  auto handle3 = task.ObserveState(LoadBundleTaskState::InProgress,
+                                   [&](LoadBundleTaskProgress p) {
+                                     EXPECT_EQ(p, Progress(1, 5));
+                                     queue.push(3);
+                                   });
+
+  task.UpdateProgress(Progress(1, 5));
+
+  EXPECT_EQ(1, queue.pop());
+  EXPECT_EQ(2, queue.pop());
+  EXPECT_EQ(3, queue.pop());
+}
+
+TEST(LoadBundleTaskTest, RemovesAllObservers) {
+  LoadBundleTask task(CreateUserQueue());
+
+  auto handle1 = task.ObserveState(LoadBundleTaskState::InProgress,
+                                   [&](LoadBundleTaskProgress p) {
+                                     (void)p;
+                                     FAIL() << "Removed observer is called.";
+                                   });
+  auto handle2 = task.ObserveState(LoadBundleTaskState::InProgress,
+                                   [&](LoadBundleTaskProgress p) {
+                                     (void)p;
+                                     FAIL() << "Removed observer is called.";
+                                   });
+  auto handle3 = task.ObserveState(LoadBundleTaskState::Error,
+                                   [&](LoadBundleTaskProgress p) {
+                                     (void)p;
+                                     FAIL() << "Removed observer is called.";
+                                   });
+  task.RemoveAllObservers();
+
+  task.UpdateProgress(Progress(1, 5));
+  task.SetError();
+
+  BlockingQueue<int> queue;
+  auto handle4 = task.ObserveState(LoadBundleTaskState::Success,
+                                   [&](LoadBundleTaskProgress p) {
+                                     EXPECT_EQ(p, SuccessProgress());
+                                     queue.push(1);
+                                   });
+  task.SetSuccess(SuccessProgress());
+
+  EXPECT_EQ(1, queue.pop());
+}
+
+TEST(LoadBundleTaskTest, ProgressesFireInOrder) {
+  LoadBundleTask task(CreateUserQueue());
+  BlockingQueue<int> queue;
+  task.ObserveState(LoadBundleTaskState::InProgress,
+                    [&](LoadBundleTaskProgress p) {
```

Comment:
If you don't use `p`, you may simply omit the name from the parameter declaration. It would avoid the need to cast to void:
```cpp
[&](LoadBundleTaskProgress) {
  //...
}
```

Reply:
Done.

Reply:
Still seems to be left in some places (e.g. line 158).

Reply:
Done